### ISSUE TITLE
Using `--buttonstyle center` causes `--button1disabled` to be ignored

### ISSUE THREAD

**Issue by mm2270** [swiftDialog/swiftDialog#300]
**Describe the bug**
Uncertain if this is a bug, or by design, but I _think_ it's a bug. Using version **2.3.1.4721** of swiftDialog, if I use `--buttonstyle center` and `--button1disabled` together to create a new dialog window, the main button is not disabled. If I just remove `--buttonstyle center`, then button 1 is disabled as expected.

**To Reproduce**

Run this script using swiftDialog version 2.3.1.4721.

```
#!/bin/zsh

dialog --buttonstyle center --button1disabled
dialog --button1disabled &
sleep 5; echo "button1: enable" >> /private/var/tmp/dialog.log
```

First dialog will come up with button1 enabled, and the button centered, even though I specified that the button be disabled. Second dialog will come up properly with button 1 disabled and the button on the right.

**Expected behavior**
I would expect that the button style being set to center wouldn't remove the ability to come up with the button disabled, since the button is still actually visible. Unless this is expected behavior since having the button centered can sometimes cause it to not show up when other options are used in the dialog.

**Screenshots**
I will add screenshots if for some reason you can't reproduce the same behavior.

**Desktop (please complete the following information):**
 - macOS Monterey, 12.6.7
 - swiftDialog version: 2.3.1.4721

**Additional context**
I don't know if this issue was present in any of the previous 2.3.x versions of swiftDialog. I wasn't using it with the `--buttonstyle` option that much. And again, it may be intentional. If it is, I apologize as I then must have missed that in the release notes description.

**Comment by bartreardon** [comment]
`--buttonstyle` is new in 2.3 and as you've discovered, no you can't disable it in the same way as you can with a regular dialog. As much as I'm against having the ability to disable the primary button at all, I'm doubly so if it's centered.

Centered buttons were done to facilitate the `--style centered` option in a general way but are intended for those small alert style dialogs and having disabled buttons in that context doesn't make sense. 

That all said, if you don't want people to activate a centered button, set the text to "none" and it will hide it entirely.

**Comment by mm2270** [comment]
Thanks for the explanation @bartreardon. As noted in my comments, as I was writing this up, I wondered if it was intentional that it couldn't be disabled.

For some context of what I was trying to achieve when I ran into this, I had a small dialog generated with an indeterminate progress bar as some other process was running in the background. This was to show that "something" was happening and to wait until it was completed. I decided on having the content centered and the button disabled so it wouldn't activate on return/enter, and so the user wouldn't be able to reflexively click it.
When the background process completed, I was using the command file to re-enable the button and change some of the content of the window (and the button label). At this point, they could close the dialog by clicking the button.

That said, I do understand the hesitancy to have disabled buttons. It's an exercise left to the admin to correctly re-enable the button or to otherwise make certain the undismissable dialog is closed in some way. Not everyone is so careful when scripting things out. So I get it.

I'm fine with using the regular right aligned buttons for this. I may also consider using what you suggested of having the button text set to "none" to see if that gets me closer to what I had wanted.

Thanks! This can be closed.
